pkg/hintrunner/hint.go

```diff
@@ -87,3 +88,155 @@ func (hint TestLessThan) Execute(vm *VM.VirtualMachine) error {
 
 	return nil
 }
+
+type DivMod struct {
+	lhs       ResOperander
+	rhs       ResOperander
+	quotient  CellRefer
+	remainder CellRefer
+}
+
+func (hint DivMod) Execute(vm *VM.VirtualMachine) error {
+
+}
+
+type Uint256DivMod struct {
+	Dividend0  ResOperander
+	Dividend1  ResOperander
+	Divisor0   ResOperander
+	Divisor1   ResOperander
+	Quotient0  CellRefer
+	Quotient1  CellRefer
+	Remainder0 CellRefer
+	Remainder1 CellRefer
+}
+
+func (hint Uint256DivMod) String() string {
+	return "Uint256DivMod"
+}
+
+func (hint Uint256DivMod) Execute(vm *VM.VirtualMachine) error {
+
+	u128MaxFelt := MaxU128Felt()
+
+	dividend0, err := hint.Dividend0.Resolve(vm)
+	if err != nil {
+		return fmt.Errorf("resolve dividend0 operand %s: %v", hint.Dividend0, err)
+	}
+
+	dividend1, err := hint.Dividend1.Resolve(vm)
+	if err != nil {
+		return fmt.Errorf("resolve dividend1 operand %s: %v", hint.Dividend1, err)
+	}
+
+	divisor0, err := hint.Divisor0.Resolve(vm)
+	if err != nil {
+		return fmt.Errorf("resolve divisor0 operand %s: %v", hint.Divisor0, err)
+	}
+
+	divisor1, err := hint.Divisor1.Resolve(vm)
+	if err != nil {
+		return fmt.Errorf("resolve divisor1 operand %s: %v", hint.Divisor1, err)
+	}
+
+	dividend0Felt, err := dividend0.ToFieldElement()
+	if err != nil {
+		return err
+	}
+
+	dividend1Felt, err := dividend1.ToFieldElement()
+	if err != nil {
+		return err
+	}
+
+	divisor0Felt, err := divisor0.ToFieldElement()
+	if err != nil {
+		return err
+	}
+
+	divisor1Felt, err := divisor1.ToFieldElement()
+	if err != nil {
+		return err
+	}
+
+	if dividend0Felt.Cmp(u128MaxFelt) > 0 {
+		return fmt.Errorf("dividend0 operand %s should be u128", dividend0Felt)
+	}
+
+	if dividend1Felt.Cmp(u128MaxFelt) > 0 {
+		return fmt.Errorf("dividend1 operand %s should be u128", dividend1Felt)
+	}
+
+	if divisor0Felt.Cmp(u128MaxFelt) > 0 {
+		return fmt.Errorf("divisor0 operand %s should be u128", divisor0Felt)
+	}
+
+	if divisor1Felt.Cmp(u128MaxFelt) > 0 {
+		return fmt.Errorf("divisor1 operand %s should be u128", divisor1Felt)
+	}
+
+	dividend := big.NewInt(1).Add(dividend0Felt.BigInt(big.NewInt(1)), dividend1Felt.BigInt(big.NewInt(1)))
+	divisor := big.NewInt(1).Add(divisor0Felt.BigInt(big.NewInt(1)), divisor1Felt.BigInt(big.NewInt(1)))
+
+	quotient, remainder := big.NewInt(1).DivMod(dividend, divisor, dividend)
+	mask := MaxU128()
+
+	quotientLow := big.NewInt(1)
+	quotientHigh := big.NewInt(1)
+	quotientLow.And(quotient, mask)
+	quotientHigh.Rsh(quotient, 128)
+
+	remainderLow := big.NewInt(1)
+	remainderHigh := big.NewInt(1)
+	remainderLow.And(remainder, mask)
+	remainderHigh.Rsh(remainder, 128)
```

Comment:
Could we perform this operations with F.Element directly. Their implementation is incredibely fast. The performance cost comes form converting to felt and viceversa, because they are stored in Montgomery Form, which is very good to perform operations modulus smth, including division, which is in constant time.

Reply:
I'm a little confused on this now, f.Element.Div just returns an f.Element and no remainder
and theres no .And or .Rsh so im not sure how to set low and high, do you have any thoughts on this?

Reply:
Yes, it makes sense. The thing is `big.Int` is made for arbitrary big numbers, but we know that for this case our numbers are 256 bits in sized. That info can help us gain a little bit of performance.

Could you use instead this [uint256 library](https://github.com/holiman/uint256). There you could see a benchmarks and it beats go native `big.Int`, also I believe it has all the same methods and more. 
If that is too much hassle, please add a `todo` there saying smth along the lines of _"swap big int usage for a more performant equivalent"_ and I'll put it there later